### PR TITLE
Character Creation Tweak

### PR DESCRIPTION
**Pull request by Geevies**
* Allows characters to set their pronouns in character creation if desired. Only affects examining and gender appearance in visible messages.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Review by Doxxmedearly — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

**Review by Arrow768 — CHANGES_REQUESTED** [review]
Merge master and update the version of the SQL migration

**Comment by NonQueueingMatt** [comment]
As I said, opposed to match gender being a thing. It should just change with the sex.

**Review by Arrow768 — CHANGES_REQUESTED** [review]
As discussed on discord:
The appearance editor needs to be updated to receive the pronoun functionality aswell.

**Comment by Geevies** [comment]
last commit removed the need to send the proc a location or a state, as it'll attach to the vueui_module and use interactive state, otherwise it runtimes because the location it gets assigned to cannot send the vueui data.

it also changes some references from "gender" to "sex", as it'll probably be an issue down the road and it's easier to nip it in the bud right now.

File: code/modules/nano/modules/human_appearance.dm [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
 	var/list/valid_species = list()
-	var/list/valid_hairstyles = list()
-	var/list/valid_facial_hairstyles = list()
-	var/list/valid_accents = list()
+	var/list/valid_genders
+	var/list/valid_pronouns
+	var/list/valid_hairstyles
+	var/list/valid_facial_hairstyles
+	var/list/valid_accents
 	var/list/valid_languages = list()
```
Comment (Arrow768) [inline code/modules/nano/modules/human_appearance.dm:12]: Style inconsistency.

File: code/modules/nano/modules/human_appearance.dm [hunk 2]
```diff
@@ -114,104 +129,131 @@
...
+	data["owner_gender"] = owner.gender
+	data["owner_pronouns"] = owner.pronouns
 	data["change_gender"] = can_change(APPEARANCE_GENDER)
-	data["accent"] = owner.accent
-	data["change_accent"] = can_change(APPEARANCE_GENDER)
+	data["valid_gender"] = null
+	data["valid_pronouns"] = null
+	if(data["change_gender"])
+		data["valid_gender"] = valid_genders
+		if(LAZYLEN(valid_pronouns))
```
Comment (Arrow768) [inline code/modules/nano/modules/human_appearance.dm:161]: Invalid usage of LAZYLEN here

File: code/game/objects/structures/mirror.dm [hunk 3]
```diff
@@ -19,7 +19,7 @@
 			to_chat(user, "<span class='notice'>Your reflection appears distorted on the surface of \the [src].</span>")
 
 	if(ishuman(user))
-		var/datum/nano_module/appearance_changer/AC = ui_users[user]
+		var/datum/vueui_module/appearance_changer/AC = ui_users[user]
```
Comment (Arrow768) [inline code/game/objects/structures/mirror.dm:22]: This looks like oldcode.

Please use the "current" way of interacting with UIs (`ui_interact` proc with `SSvueui-get_open_ui()`)

File: code/game/objects/structures/mirror.dm [hunk 4]
```diff
@@ -71,7 +71,7 @@
 
 /obj/structure/mirror/Destroy()
 	for(var/user in ui_users)
-		var/datum/nano_module/appearance_changer/AC = ui_users[user]
+		var/datum/vueui_module/appearance_changer/AC = ui_users[user]
```
Comment (Arrow768) [inline code/game/objects/structures/mirror.dm:74]: Same here

File: code/game/objects/structures/mirror.dm [hunk 5]
```diff
@@ -90,7 +90,7 @@
 			to_chat(user, "<span class='notice'>Your reflection appears distorted on the surface of \the [src].</span>")
 
 	if(ishuman(user))
-		var/datum/nano_module/appearance_changer/AC = ui_users[user]
+		var/datum/vueui_module/appearance_changer/AC = ui_users[user]
```
Comment (Arrow768) [inline code/game/objects/structures/mirror.dm:93]: Same here

File: code/game/objects/structures/mirror.dm [hunk 6]
```diff
@@ -100,7 +100,7 @@
 
 /obj/item/mirror/Destroy()
 	for(var/user in ui_users)
-		var/datum/nano_module/appearance_changer/AC = ui_users[user]
+		var/datum/vueui_module/appearance_changer/AC = ui_users[user]
```
Comment (Arrow768) [inline code/game/objects/structures/mirror.dm:103]: Same here

File: code/modules/mob/living/carbon/human/appearance.dm [hunk 7]
```diff
@@ -1,7 +1,7 @@
-/mob/living/carbon/human/proc/change_appearance(var/flags = APPEARANCE_ALL_HAIR, var/location = src, var/mob/user = src, var/check_species_whitelist = 1, var/list/species_whitelist = list(), var/list/species_blacklist = list(), var/datum/topic_state/state = default_state)
-	var/datum/nano_module/appearance_changer/AC = new(location, src, check_species_whitelist, species_whitelist, species_blacklist)
+/mob/living/carbon/human/proc/change_appearance(var/flags = APPEARANCE_ALL_HAIR, var/mob/user = src, var/check_species_whitelist = 1, var/list/species_whitelist = list(), var/list/species_blacklist = list())
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/appearance.dm:1]:
```suggestion
/mob/living/carbon/human/proc/change_appearance(var/flags = APPEARANCE_ALL_HAIR, var/mob/user = src, var/check_species_whitelist = TRUE, var/list/species_whitelist = list(), var/list/species_blacklist = list())
```

**Review by Arrow768 — CHANGES_REQUESTED** [review]

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**